File: androidApp/src/main/kotlin/fr/abknative/outgo/android/screens/SettingsScreen.kt
package fr.abknative.outgo.android.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.SnackbarHost
import androidx.compose.material3.SnackbarHostState
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.text.style.TextAlign
import fr.abknative.outgo.android.R
import fr.abknative.outgo.android.components.common.ConfirmationDialog
import fr.abknative.outgo.android.components.common.HoldToConfirmButton
import fr.abknative.outgo.android.components.common.SecondaryButton
import fr.abknative.outgo.android.components.settings.*
import fr.abknative.outgo.android.ui.CommonLabels
import fr.abknative.outgo.android.ui.DialogLabels
import fr.abknative.outgo.android.ui.SettingsLabels
import fr.abknative.outgo.android.ui.theme.AppTheme
import fr.abknative.outgo.android.ui.theme.toColor
import fr.abknative.outgo.android.ui.toUIString
import fr.abknative.outgo.settings.api.SettingsIntent
import fr.abknative.outgo.settings.api.SettingsPresenter

@Composable
fun SettingsScreen(
    presenter: SettingsPresenter,
    onNavigateToLogin: () -> Unit,
    isDarkMode: Boolean,
    onToggleDarkMode: (Boolean) -> Unit,
    modifier: Modifier = Modifier
) {
    val state by presenter.state.collectAsState()
    val scrollState = rememberScrollState()
    val uriHandler = LocalUriHandler.current

    val snackbarHostState = remember { SnackbarHostState() }
    val currentError = state.error
    val errorMessage = currentError?.toUIString()

    LaunchedEffect(currentError) {
        if (currentError != null && errorMessage != null) {
            snackbarHostState.showSnackbar(message = errorMessage, withDismissAction = true)
            presenter.onIntent(SettingsIntent.DismissError)
        }
    }

    LaunchedEffect(state.actionSuccess) {
        if (state.actionSuccess) {
            presenter.onIntent(SettingsIntent.ResetSuccessFlag)
        }
    }

    // --- Dialog States ---
    var showLogoutOptions by remember { mutableStateOf(false) }
    var showDeleteAccountConfirm by remember { mutableStateOf(false) }
    var showPurgeConfirm by remember { mutableStateOf(false) }

    Box(modifier = modifier.fillMaxSize()) {

        // Le gestionnaire de notifications
        SnackbarHost(
            hostState = snackbarHostState,
            modifier = Modifier.align(Alignment.TopCenter)
        )

        Column(
            modifier = modifier
                .fillMaxSize()
                .verticalScroll(scrollState)
                .padding(AppTheme.spacing.medium),
            verticalArrangement = Arrangement.spacedBy(AppTheme.spacing.large)
        ) {

            // --- SECTION Apparence ---
            SettingsSection(title = SettingsLabels.SECTION_APPEARANCE) {
                SettingsRowToggle(
                    icon = R.drawable.moon_duotone,
                    title = SettingsLabels.DARK_MODE_TITLE,
                    subtitle = SettingsLabels.DARK_MODE_SUBTITLE,
                    isChecked = isDarkMode,
                    onCheckedChange = onToggleDarkMode
                )
            }

            // --- SECTION Soutien & Communauté ---
            SettingsSection(title = SettingsLabels.SECTION_SUPPORT) {
                SettingsRowClickable(
                    icon = R.drawable.lightbulb_duotone,
                    title = SettingsLabels.TIPS_TITLE,
                    subtitle = SettingsLabels.TIPS_SUBTITLE,
                    onClick = { uriHandler.openUri(SettingsLabels.URL_SITE) }
                )
                HorizontalDivider(color = AppTheme.colors.textSecondary.toColor().copy(alpha = 0.1f))
                SettingsRowClickable(
                    icon = R.drawable.envelope_duotone,
                    title = SettingsLabels.CONTACT_TITLE,
                    subtitle = SettingsLabels.CONTACT_SUBTITLE,
                    onClick = { uriHandler.openUri(SettingsLabels.URL_CONTACT) }
                )
            }

            // --- SECTION Données & Compte ---
            if (state.session == null) {
                SettingsSection(title = SettingsLabels.SECTION_DATA_AND_ACCOUNT) {
                    SettingsRowClickable(
                        icon = R.drawable.arrows_clockwise_duotone,
                        title = SettingsLabels.SYNC_TITLE,
                        subtitle = SettingsLabels.SYNC_SUBTITLE,
                        onClick = { onNavigateToLogin() }
                    )
                    HorizontalDivider(color = AppTheme.colors.textSecondary.toColor().copy(alpha = 0.1f))
                    SettingsRowClickable(
                        icon = R.drawable.trash_duotone,
                        title = SettingsLabels.PURGE_TITLE,
                        subtitle = SettingsLabels.PURGE_SUBTITLE,
                        onClick = { showPurgeConfirm = true }
                    )
                }
            } else {
                SettingsSection(title = SettingsLabels.SECTION_ACCOUNT) {
                    SettingsRowClickable(
                        icon = R.drawable.sign_out_duotone,
                        title = SettingsLabels.LOGOUT_TITLE,
                        subtitle = SettingsLabels.LOGOUT_SUBTITLE,
                        onClick = { showLogoutOptions = true }
                    )
                    HorizontalDivider(color = AppTheme.colors.textSecondary.toColor().copy(alpha = 0.1f))
                    SettingsRowClickable(
                        icon = R.drawable.trash_duotone,
                        title = SettingsLabels.DELETE_ACCOUNT_TITLE,
                        subtitle = SettingsLabels.DELETE_ACCOUNT_SUBTITLE,
                        onClick = { showDeleteAccountConfirm = true }
                    )
                }
            }

            // Footer
            Text(
                text = SettingsLabels.APP_VERSION_PREFIX,
                style = AppTheme.typo.caption,
                color = AppTheme.colors.textPrimary.toColor(),
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(top = AppTheme.spacing.large),
                textAlign = TextAlign.Center
            )
        }

        // --- MODAL DE CONFIRMATION ---
        if (showLogoutOptions) {
            LogoutDialog(
                onKeepOffline = {
                    // On reste sur l'ID actuel (Firebase ID mais offline)
                    presenter.onIntent(SettingsIntent.Logout(displayLocalData = true))
                    showLogoutOptions = false
                },
                onReturnToLocal = {
                    // On bascule sur le vieil ID local
                    presenter.onIntent(SettingsIntent.Logout(displayLocalData = false))
                    showLogoutOptions = false
                },
                onCancel = {
                    showLogoutOptions = false
                }
            )
        }

        if (showDeleteAccountConfirm) {
            DeleteAccountDialog(
                onConfirm = { wipeLocal, wipeServer, revokeAuth ->
                    presenter.onIntent(SettingsIntent.DeleteAccount(wipeLocal, wipeServer, revokeAuth))
                    showDeleteAccountConfirm = false
                },
                onDismiss = { showDeleteAccountConfirm = false }
            )
        }

        if (showPurgeConfirm) {
            ConfirmationDialog(
                title = DialogLabels.PURGE_TITLE,
                description = DialogLabels.PURGE_DESC,
                onDismiss = { showPurgeConfirm = false },
                confirmButton = {
                    HoldToConfirmButton(
                        label = DialogLabels.PURGE_CONFIRM,
                        onConfirm = {
                            presenter.onIntent(SettingsIntent.PurgeLocalData)
                            showPurgeConfirm = false
                        }
                    )
                },
                dismissButton = {
                    SecondaryButton(
                        label = CommonLabels.ACTION_CANCEL,
                        labelColor = AppTheme.colors.textSecondary.toColor(),
                        onClick = { showPurgeConfirm = false },
                        modifier = Modifier.padding(end = AppTheme.spacing.medium)
                    )
                }
            )
        }
    }
}

File: iosApp/iosApp/screens/SettingsScreen.swift
import SwiftUI
import SharedApp

struct SettingsScreen: View {
    // --- Propriétés ---
    let onNavigateBack: () -> Void
    let isDarkMode: Bool
    let onToggleDarkMode: (Bool) -> Void
    let onCoffeeClick: () -> Void
    let onTipsClick: () -> Void
    let onContactClick: () -> Void
    
    // --- État Local ---
    @State private var showSyncModal = false
    
    // --- Environnement ---
    @Environment(\.spacing) private var spacing
    @Environment(\.outgoColors) private var colors
    @Environment(\.outgoTypography) private var typo
    
    var body: some View {
        ZStack {
            colors.background.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // --- 1. Header ---
                Header(
                    isConnected: false, // À lier à l'Auth plus tard
                    isSettingsScreen: true,
                    onSyncIconClick: { showSyncModal = true },
                    onSyncNavigationClick: onNavigateBack
                )
                
                ScrollView {
                    VStack(spacing: spacing.large) {
                        
                        // --- SECTION 1 : Apparence ---
                        SettingsSection(title: SettingsLabels.shared.SECTION_APPEARANCE) {
                            SettingsRowToggle(
                                icon: "moon_duotone",
                                title: SettingsLabels.shared.DARK_MODE_TITLE,
                                subtitle: SettingsLabels.shared.DARK_MODE_SUBTITLE,
                                isChecked: Binding(
                                    get: { isDarkMode },
                                    set: { onToggleDarkMode($0) }
                                )
                            )
                        }
                        
                        // --- SECTION 2 : Soutien & Communauté ---
                        SettingsSection(title: SettingsLabels.shared.SECTION_SUPPORT) {
                            SettingsRowClickable(
                                icon: "lightbulb_duotone",
                                title: SettingsLabels.shared.TIPS_TITLE,
                                subtitle: SettingsLabels.shared.TIPS_SUBTITLE,
                                onClick: onTipsClick
                            )
                            
                            customDivider
                            
                            SettingsRowClickable(
                                icon: "envelope_duotone",
                                title: SettingsLabels.shared.CONTACT_TITLE,
                                subtitle: SettingsLabels.shared.CONTACT_SUBTITLE,
                                onClick: onContactClick
                            )
                            
                            customDivider
                            
                            SettingsRowClickable(
                                icon: "coffee_duotone",
                                title: SettingsLabels.shared.COFFEE_TITLE,
                                subtitle: SettingsLabels.shared.COFFEE_SUBTITLE,
                                onClick: {
                                    // onCoffeeClick()
                                }
                            )
                        }
                        
                        // --- SECTION 3 : Compte & Données ---
                        SettingsSection(title: SettingsLabels.shared.SECTION_DATA) {
                            SettingsRowClickable(
                                icon: "arrows_clockwise_duotone",
                                title: SettingsLabels.shared.SYNC_TITLE,
                                subtitle: SettingsLabels.shared.SYNC_SUBTITLE,
                                onClick: { showSyncModal = true }
                            )
                        }
                        
                        // Version de l'app (Footer)
                        Text(SettingsLabels.shared.APP_VERSION_PREFIX)
                            .font(typo.caption)
                            .foregroundColor(colors.textSecondary)
                            .padding(.top, spacing.large)
                    }
                    .padding(spacing.medium)
                }
            }
        }
        // --- 2. Modale de Promotion Sync ---
        .syncPromotionModal(isPresented: $showSyncModal) {
            showSyncModal = false
            // TODO: Navigation vers Auth
        }
    }
    
    private var customDivider: some View {
        Divider().background(colors.textSecondary.opacity(0.1))
    }
}

// --- Previews ---
#Preview("Mode Clair") {
    SettingsScreen(
        onNavigateBack: {},
        isDarkMode: false,
        onToggleDarkMode: { _ in },
        onCoffeeClick: {},
        onTipsClick: {},
        onContactClick: {}
    )
}
